lib/engine/round/g_1822/stock.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+require_relative '../stock'
+
+module Engine
+  module Round
+    module G1822
+      class Stock < Stock
+        attr_accessor :bids, :bidders
+
+        def finish_round
+          @game.bidbox_minors.map do |minor_company|
+            bid = highest_bid(minor_company)
+            float_minor(bid) unless bid.nil?
+          end
+
+          @game.bidbox_concessions.map do |concessions|
+            bid = highest_bid(concessions)
+            buy_company(bid) unless bid.nil?
+          end
+
+          @game.bidbox_privates.map do |private_company|
+            bid = highest_bid(private_company)
+            buy_company(bid) unless bid.nil?
+          end
+
+          super
+        end
+
+        def setup
+          super
+        end
+
+        def buy_company(bid)
+          player = bid.entity
+          company = bid.company
+          price = bid.price
+
+          company.owner = player
+          player.companies << company
+          player.spend(price, @game.bank) if price.positive?
+          @game.companies.delete(company)
+          @log << "#{player.name} buys #{company.name} for #{@game.format_currency(price)}"
+        end
+
+        def float_minor(bid)
+          player = bid.entity
+          company = bid.company
+          price = bid.price
+
+          return unless (minor = @game.minors.find { |m| m.id == company.id })
+
+          player.spend(price, @game.bank)
+
+          minor.owner = player
+          minor.float!
+          @game.bank.spend(100, minor)
+
+          hex = @game.hex_by_id(minor.coordinates)
+          hex.tile.cities[minor.city || 0].place_token(minor, minor.next_token)
```

Comment:
If I recall correctly, token placement is part of first turn housekeeping. It shouldn't block another minor that runs earlier from running through the city.

Reply:
Thats true, but as you see in the rest of the method par price and other stuff is hardcoded. It is not finished, just places the token in that part of the code just not to break the game.

Reply:
I add TODO comments for those types of things, so I don't forget. That's just a personal preference though.

Reply:
Im not used to add TODO that much. But since its hard for the reviews to know how i think i might add a bit more todo's code that is not finished. But in this state most of the code isnt finished :)

Reply:
Sure. If you don't like the TODOs, including comments in the PR can provide context to reviewers too. Whatever works best for you and your workflow. :-)